OWML.ModHelper/Logging/ModSocketOutput.cs

```diff
@@ -20,34 +20,79 @@ public ModSocketOutput(IOwmlConfig config, IModLogger logger, IModManifest manif
             if (_socket == null)
             {
                 _port = config.SocketPort;
-                config.SocketPort = -1;
-                JsonHelper.SaveJsonObject(Constants.OwmlConfigFileName, config);
-                CreateSocket();
+                ConnectToSocket();
             }
         }
 
-        public override void WriteLine(string s)
+        [Obsolete]
+        public override void WriteLine(params object[] objects)
         {
-            Logger?.Log(s);
-            CallWriteCallback(Manifest, s);
-            var message = $"{Manifest.Name}{NameMessageSeparator}{s}";
-            InternalWriteLine(message);
+            var line = string.Join(" ", objects.Select(o => o.ToString()).ToArray());
+            MessageType type;
+            if (line.ToLower().Contains("error") || line.ToLower().Contains("exception"))
```

Comment:
not you who created this horrible if-else, but: 
you could extract one of the ifs to a new private method and simplify it:
```
private bool Contains(string line, params string[] keyWords)
{
    return keyWords.Any(keyWord => line.Contains);
}
```

Reply:
After doing that, maybe VS will suggest something even shorter

Reply:
and do `var type = GetType(line);`